source/AI.cpp

```diff
@@ -1235,9 +1240,11 @@ shared_ptr<Ship> AI::FindTarget(const Ship &ship) const
 				if(it->GetGovernment() != gov)
 				{
 					auto ptr = it->shared_from_this();
-					// Scan friendly ships that are as-yet unscanned by this ship's government.
-					if((!cargoScan || Has(gov, ptr, ShipEvent::SCAN_CARGO))
+					// Scan friendly ships that are as-yet unscanned by this ship's government,
+					// and are not in the process of being scanned.
```

Comment:
```suggestion
					// and are not in the process of being scanned, except if this is the ship doing it.
```
Small clarification

But actually this whole comment is repeated, and should be a comment of the method I describe in the next github comment

Reply:
```suggestion
					// and are not in the process of being scanned by another ship in this government.
```